extensions/mssql/src/errorDiagnostics/errorDiagnosticsProvider.ts

```diff
@@ -51,13 +51,20 @@ export class ErrorDiagnosticsProvider extends SqlOpsFeature<any> {
 			}
 
 			protected override registerProvider(options: any): Disposable {
-				let handleConnectionError = async (errorCode: number, errorMessage: string, connection: azdata.connection.ConnectionProfile): Promise<azdata.diagnostics.ConnectionDiagnosticsResult> => {
-					let restoredProfile = this.convertToIConnectionProfile(connection);
-					if (errorCode === ErrorDiagnosticsConstants.MssqlPasswordResetErrorCode) {
-						logDebug(`Error Code ${errorCode} requires user to change their password, launching change password dialog.`)
-						return await this.handleChangePassword(restoredProfile);
+				let handleConnectionError = async (errorInfo: azdata.diagnostics.IErrorInformation, connection: azdata.connection.ConnectionProfile): Promise<azdata.diagnostics.ConnectionDiagnosticsResult> => {
+					if (errorInfo) {
```

Comment:
nit: Do you actually expect this to be undefined ever? We control sending this so I think it's fine to assume it'll always be defined